daemon/algod/api/server/v2/handlers_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package v2
+
+import (
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"github.com/stretchr/testify/require"
+	"math"
+	"math/rand"
+	"testing"
+)
+
+func TestApplicationBoxesMaxKeys(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	type example struct {
+		requestedMax uint64
+		algodMax     uint64
+	}
+
+	randomUint64 := func(min, max uint64) uint64 {
+		r := rand.Uint64()
+		if r > max {
+			return max
+		} else if r < min {
+			return min
+		}
+		return r
+	}
+
+	equals := func(expected uint64, e example) {
+		require.Equal(t, expected, applicationBoxesMaxKeys(e.requestedMax, e.algodMax), "failing example = %+v", e)
+	}
+
+	// Response size limited by request supplied value.
+	{
+		requestedMax := randomUint64(1, math.MaxUint64-1)
```

Comment:
Unless there is a strong reason for randomness, I would discourage introducing it here. In this case, it shouldn't matter but it still introduces uncertainty without any benefit (at least not one that I can see). If we ever get a failing test it would seem to indicate a flake. And if it isn't a flake, then it argues that the test should have been designed more carefully to test the edge case that was discovered by chance.

Reply:
@tzaffi Feedback addressed in https://github.com/algorand/go-algorand/pull/4543/commits/2aafae80c0fd3ab55e31719b48045b0083a68145.